DeviceAdministration/Infrastructure/BusinessLogic/AlertsLogic.cs

```diff
@@ -36,13 +36,24 @@ public AlertsLogic(IAlertsRepository alertsRepository)
         /// <param name="minTime">
         /// The cutoff time for Device Alert History items that should be returned.
         /// </param>
+        /// <param name="minResults">
+        /// The minimum number of items that should be returned, if possible, 
+        /// after <paramref name="minTime"/> or otherwise.
+        /// </param>
         /// <returns>
         /// The latest Device Alert History items.
         /// </returns>
-        public async Task<IEnumerable<AlertHistoryItemModel>> LoadLatestAlertHistoryAsync(DateTime minTime)
+        public async Task<IEnumerable<AlertHistoryItemModel>> LoadLatestAlertHistoryAsync(DateTime minTime, int minResults)
         {
+            if (minResults <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
```

Comment:
Yes--for invalid data, the REST API would return an error, and the dashboard would display a dialog box saying it could not load the alerts.

Currently this is only populated from a positive constant.

Our plan here is to identify invalid data and then "fail fast". Did you have another strategy you would prefer?